src/models/cards/path-cards/boundary-cards.ts

```diff
@@ -25,7 +25,7 @@ export class FinishPathCard extends PathCard {
 
   turnOver(side: Sides): void {
     this.isFaceDown = false;
-    if (!this.connectors.includes(side)) {
+    if (!this.connectors.includes(side) && !this.isUpsideDown) {
```

Comment:
If the card is right side up and doesn't connect, rotate it. If it's upside down then it'll already connect, no need to rotate!